src/App.js

```diff
@@ -1,33 +1,61 @@
 // Packages
-import React from 'react';
+import { useState, useEffect } from 'react';
 import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
 
+// Firestore
+import { db } from './lib/firebase.js';
+
 // External Files
 import './App.css';
 
 // View Components
-import List from './pages/List/List';
-import AddItem from './pages/AddItem/AddItem';
+import ListView from './pages/ListView/ListView';
+import AddItemView from './pages/AddItemView/AddItemView';
 import NotFound from './pages/NotFound/NotFound';
 
-// Components
-import NavMenu from './components/NavMenu/NavMenu';
-
 function App() {
+  const [listId, setListId] = useState(null);
+
+  // on component mounting, look for token in local storage and use it to retrieve the list id
+  useEffect(() => {
+    const token = localStorage.getItem('token');
+
+    if (token) {
+      // find the list in Firestore associated with the stored token from local storage
+      db.collection('lists')
+        .where('token', '==', token)
+        .get()
+        .then((querySnapshot) => {
+          // for some reason, extracting this array seems necessary to avoid errors?
+          const documents = querySnapshot.docs;
+
+          // check if the expected data is returned
+          if (
+            Array.isArray(documents) &&
+            documents.length &&
+            typeof documents[0] === 'object'
+          ) {
+            setListId(documents[0].id); // save the list id for later
```

Comment:
I really like how thorough this check is! @nick-zanetti  I believe the check for length is to ensure that we are not getting back an empty array. It could be argued that the next line makes that check as well by assuming that there is a value at the index of 0, but personally I like how explicit line 35 is about the check

Reply:
1. Is it an array?
2. Is there anything in the array? (length != 0)
3. Is the first (and probably only) thing in the array an object

...to be even more safe could probably add:
4. Does the object even have an id property?

The thinking is doing these checks, in this order, will hopefully prevent getting errors if something doesn't exist in the database or there's some other problem with results returned. Just straight up referring to `documents[0].id` if I didn't get back data in the format I'm expecting maybe not go over well. Though honestly, I haven't worked enough with firebase or had time yet to test different possibilities and learn the format of the result I'd get in those cases. 

This is likely a habit I'm in from coding in PHP, where if you make an assumption like that and it doesn't pan out, your entire page might be kaput (or at best of visible errors), so checks like this limit the failure to one area where you write in a fallback. Interested to hear from mentors if this is overkill in JavaScript?

Reply:
@nick-zanetti I played around to kind of remind myself what JavaScript will and won't let you get away with, and if you were to do something like `someVar[0]`, `someVar.id` or `someVar[0].id` on an undefined or null value, you get TypeErrors and the app crashes. So I think I'm going to quickly add in that extra hasOwnProperty check as well!